src/consumer.rs

```diff
@@ -37,6 +37,45 @@ impl<
     }
 }
 
+/// Continuously consumes message from a Queue
+///
+/// A consumer represents the [basic.consume](https://www.rabbitmq.com/amqp-0-9-1-quickref.html#basic.consume) AMQP command.
+/// It continuously receives messages from the queue, as opposed to the
+/// [basic.get](https://www.rabbitmq.com/amqp-0-9-1-quickref.html#basic.get) command, which
+/// retrieves only a single message.
+///
+/// A consumer is obtained by calling [`Channel::basic_consume`] with the queue name.
+/// New messages from this consumer can be accessed by obtaining the iterator from the consumer.
```

Comment:
There is also the `set_delegate` method which automatically spawns the delegate on the executor for each message (so you basically get parallel handling of messages)